docs/guides/using-lighthouse-dns/validateDomain.ts

```diff
@@ -0,0 +1,16 @@
+export const INVALID_DOMAIN = 'Invalid domain';
+
+export function validateDomain(domain: string): true | typeof INVALID_DOMAIN {
+  if (!domain.length) {
+    return true;
+  }
+
+  // sourced from https://www.oreilly.com/library/view/regular-expressions-cookbook/9781449327453/ch08s15.html
+  const domainRegex = new RegExp('\\b((?=[a-z0-9-]{1,63}\\.)(xn--)?[a-z0-9]+(-[a-z0-9]+)*\\.)+[a-z]{2,63}\\b', 'gm');
```

Comment:
```suggestion
  const domainRegex = new RegExp('\\b((?=[a-z0-9-]{1,63}\\.)(xn--)?[a-z0-9]+(-[a-z0-9]+)*\\.)+[a-z]{2,63}\\b', 'gmi');
```

This should be a case insensitive match, per the link. Probably worth a test case too.

Perhaps also a test case around international punycode domains too?

Reply:
the test should be covered by "should validate generated domains" but I'll enter an explicit test as well. I assume that this regex is already case insensitive?

Reply:
It's only using lowercase letters, so no it's not.  The source you linked to specifies:

```
Regex options: Case insensitive
```